core/src/main/java/org/infinispan/marshall/exts/OptionalExternalizer.java

```diff
@@ -0,0 +1,38 @@
+package org.infinispan.marshall.exts;
+
+import org.infinispan.commons.marshall.AbstractExternalizer;
+import org.infinispan.commons.util.Util;
+import org.infinispan.marshall.core.Ids;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.Optional;
+import java.util.Set;
+
+public class OptionalExternalizer extends AbstractExternalizer<Optional> {
```

Comment:
Oh remember, that a return from a function might get serialized, e.g. in a clustered environment.